Content.Server/Atmos/Piping/Unary/EntitySystems/GasCanisterSystem.cs

```diff
@@ -131,11 +137,14 @@ private void OnCanisterChangeReleaseValve(EntityUid uid, GasCanisterComponent ca
         impact = canister.GasTankSlot.HasItem ? LogImpact.Medium : LogImpact.High;
 
         var containedGasDict = new Dictionary<Gas, float>();
-        var containedGasArray = Enum.GetValues(typeof(Gas));
 
-        for (int i = 0; i < containedGasArray.Length; i++)
+        if (TryComp<InternalAirComponent>(uid, out var internalAir))
         {
-            containedGasDict.Add((Gas)i, canister.Air[i]);
+            var containedGasArray = Enum.GetValues(typeof(Gas));
+            for (int i = 0; i < containedGasArray.Length; i++)
+            {
+                containedGasDict.Add((Gas)i, internalAir.Air[i]);
+            }
         }
 
         _adminLogger.Add(LogType.CanisterValve, impact, $"{ToPrettyString(args.Actor):player} set the valve on {ToPrettyString(uid):canister} to {args.Valve:valveState} while it contained [{string.Join(", ", containedGasDict)}]");
```

Comment:
you just log `{internalAir.Air:gas}` above cant you do the same here

Reply:
If that logs the same amount of information, sure.
I don't know how to toggle the release valve to test that in-game so I just refactored it to have the same behavior as it did before.

But I've changed it to the simpler logging code now.